### PR TITLE
Reserved ranges for actors and methods

### PR DESCRIPTION
**Pull request by mishmosh**
This PR covers:
- Stub out the listings sections for actors, methods, and errors
- Add Actors and Methods listings, based on [vm_registry.go](https://github.com/filecoin-project/specs/blob/master/src/systems/filecoin_vm/interpreter/vm_registry.go) and [methods.go](https://github.com/filecoin-project/specs/blob/7d174209c2a7afda755e524f760a04923bbf7b14/src/systems/filecoin_vm/actor_interfaces/methods.go)
- Move an errant `EpochTick` method from `sysactors/cron_actor.go` to `actor_interfaces/methods.go` as suggested by @sternhenri 

Error codes will be done in a future PR.

Questions for reviewers:
1. Can the "[Filecoin VM Actors](https://filecoin-project.github.io/specs/#listings__actors)" section be deleted, or was something else planned for it? 
2. For methods, removing the prefix might be cleaner. Thoughts? i.e. `Method_MultiSigActor_AddAuthorizedParty` -> `AddAuthorizedParty`
3. Missing actors/missing methods. See [comment](https://github.com/filecoin-project/specs/pull/785#issuecomment-572291348) below.

### CODE REVIEW

**Comment by sternhenri** [comment]
Looks great so far. I might add the methods under the appropriate actors, yes

**Comment by mishmosh** [comment]
Most of this is based on the `.go` files, but there are some missing listings which I intend to add.

**Listed in Actors, but missing methods:**
SystemActor
AccountActor
CronActor
PaymentChannelActor

**Listed in methods, but missing in Actors list:**
RewardActor
MultiSigActor

File: src/listings/reserved_ranges.md [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+| Actor                | ID |
+|---|---|
+| SystemActor          | 1 |
+| InitActor            | 2 |
+| CronActor            | 3 |
+| AccountActor         | 4 |
+| StoragePowerActor    | 5 |
+| StorageMinerActor    | 6 |
+| StorageMarketActor   | 7 |
+| PaymentChannelActor  | 8 |
```
Comment (anorth) [inline src/listings/reserved_ranges.md]: Call this "Actor ID" reserved ranges.

I think you may have confused BuiltinActorID actor-type enumeration which I have literally just deleted (#795) from `actor.id`, with the reserved actor _ID addresses_ defined in `address.go`. These are the ones we want to refer to here. Sorry for the confusion. There's also

```
const FirstNonSingletonActorId = 100
```
So the reserved range is all values lower than 100 for singleton actors.

File: src/listings/reserved_ranges.md [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+|	Method_StoragePowerActor_OnSectorProveCommit                  | 7 |
+|	Method_StoragePowerActor_OnSectorTemporaryFaultEffectiveBegin | 8 |
+|	Method_StoragePowerActor_OnSectorTemporaryFaultEffectiveEnd   | 9 |
+|	Method_StoragePowerActor_OnSectorModifyWeightDesc             | 10 |
+|	Method_StoragePowerActor_OnSectorTerminate                    | 11 |
+|	Method_StoragePowerActor_OnMinerSurprisePoStSuccess           | 12 |
+|	Method_StoragePowerActor_OnMinerSurprisePoStFailure           | 13 |
+|	Method_StoragePowerActor_OnMinerEnrollCronEvent               | 14 |
+|	Method_StoragePowerActor_GetMinerConsensusPower               | 15 |
+|	Method_StoragePowerActor_GetMinerUnmetPledgeCollateralRequirement | 16 |
```
Comment (anorth) [inline src/listings/reserved_ranges.md]: I'm not sure it's a good idea to repeat these here. They need to be fully specified by the actors (which will move out of this repo), but there will probably be a bit of churn and these two representations will depart.

As far as reserved ranges go, we reserve method ID `0` for value-send and `1` for constructor, and all other positive values are free for actors to use.

**Comment by mishmosh** [comment]
Ready for another look @anorth.

File: src/listings/reserved_ranges.md (hunk 2 above)
Comment (mishmosh) [reply src/listings/reserved_ranges.md]: I removed the full methods listing, and added a TODO (our favourite) to link to actor code when done.

File: src/listings/reserved_ranges.md (hunk 1 above)
Comment (mishmosh) [reply src/listings/reserved_ranges.md]: Fixed.

**Pull request merged**